### ISSUE TITLE
Field validation

### ISSUE THREAD

**Issue by quen2404** [deepmap/oapi-codegen#227]
Hi, 
thank you for your project, it's exactly what i'm looking for some time :100: 
I'm also developing rest API using Go and Echo and we use OpenAPI to document ours apis. 
In my apis, i'm a huge fan & user of package [go-playground/validator](https://github.com/go-playground/validator) to validate data i received (like min string length , etc...). 
It will be awesome if you could support this in schema structure generation. What do you think ? Do you need help to do that ?

**Comment by skateinmars** [comment]
One thing that could be great would be to use the JSON-Schema tags that can be included in the openapi schema itself
https://swagger.io/docs/specification/data-models/keywords/

Some libs already exist to use JSON Schemas like https://github.com/xeipuuv/gojsonschema so the work may not be too huge

**Comment by kolluria** [comment]
I had a similar requirement and I was able to build a viable solution by generating an echo server using oapi-codegen, and then registering a custom validator that implements go-playground/validator as mentioned in the echo [documentation](https://echo.labstack.com/guide/request/#validate-data).

This is how you could do it - 

Add a custom validate tag in the swagger for the fields that require validation
say 

```
 x-oapi-codegen-extra-tags:
            validate: excluded_with_all
```
The models get generated with this custom tag 

```
`json:"id,omitempty" validate:"excluded_with_all"`
```

**Comment by bofm** [comment]
@kolluria did you have to call `echo_ctx.Validate(myStructPointer)` manually for each handler?

**Comment by kolluria** [comment]
@bofm Yeah. You have to manually invoke `echo_ctx.Validate` after binding the request body to the struct.

For instance, 

`func (s serverImplementation) SomeHandler(c echo.Context) error {
	var body SomeJSONRequestBody
	err = c.Bind(&body)
	if err != nil {
		return err
	}

	err = c.Validate(&body)
	if err != nil {
		return err
	}
       // Your business logic goes here
}`

**Comment by bofm** [comment]
@kolluria thanks for the answer.

I just found a way to not have to (and not forget to) call Validate() in each handler with the strict server:

```go
type StructValidator struct {
    validator *validator.Validate
}

func (v StructValidator) Validate(i interface{}) error {
    if err := v.validator.Struct(i); err != nil {
        return err
    }
    return nil
}

func validateStructMiddeware(f api.StrictHandlerFunc, operationID string) api.StrictHandlerFunc {
    return func(ctx echo.Context, i interface{}) (interface{}, error) {
        if err := ctx.Validate(i); err != nil {
            return nil, fmt.Errorf("%s failed to validate request body: %w", operationID, err)
        }
        return f(ctx, i)
    }
}

func main() {
    e := echo.New()
    e.Validator = StructValidator{validator: v}

    apiServer := api.NewStrictHandler(
        api.NewServer(theStorage),
        []api.StrictMiddlewareFunc{
            validateStructMiddeware,
        },
    )
    api.RegisterHandlers(e, apiServer)
}
```

**Comment by pbikki** [comment]
this does not seem to work for nested fields. has anyone has success with tagging a nested field

**Comment by mromaszewicz** [comment]
Issue #227 requested support for go-playground/validator tags in generated structs. This is fully addressed by the existing x-oapi-codegen-extra-tags extension, which allows arbitrary struct tags (including validate:) to be added to generated fields. The README documents this with an explicit validate: example (README.md:2920-2973), and a community commenter (kolluria, 2022-12-08) already explained this exact solution in the issue thread. No code changes are needed; the feature exists and is documented.

Closing as fixed. If you can still reproduce on a recent release, please reopen with the exact spec and version used.
